recipes/libftdi/1.x/conanfile.py

```diff
@@ -10,94 +13,86 @@ class LibFtdiConan(ConanFile):
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def export_sources(self):
+        export_conandata_patches(self)
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = "libftdi1-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version],
+            strip_root=True)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
             if self.settings.compiler == "Visual Studio":
                 self.options.use_streaming = False
 
+    def layout(self):
+        cmake_layout(self)
+
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-
-        self._cmake = CMake(self)
-        options = {
-            "BUILD_TESTS": False,
-            "EXAMPLES": False,
-            "FTDI_EEPROM": self.options.build_eeprom_tool,
-            "FTDIPP" : self.options.enable_cpp_wrapper,
-            "STATICLIBS": not self.options.shared,
-            "ENABLE_STREAMING": self.options.use_streaming,
-        }
-        self._cmake.definitions.update(options)
-        self._cmake.configure()
-        return self._cmake
+    def generate(self):
+        tc = CMakeToolchain(self)
+        # Boolean values are preferred instead of "ON"/"OFF"
+        tc.variables["BUILD_TESTS"] = False
+        tc.variables["EXAMPLES"] = False
+        tc.variables["FTDI_EEPROM"] = self.options.build_eeprom_tool
+        tc.variables["FTDIPP"] = self.options.enable_cpp_wrapper
+        tc.variables["ENABLE_STREAMING"] = self.options.use_streaming
+        tc.variables["LIB_SUFFIX"] = ""
+        tc.generate()
+        # In case there are dependencies listed on requirements, CMakeDeps should be used
+        tc = CMakeDeps(self)
+        tc.generate()
 
     def requirements(self):
         self.requires("libusb/1.0.24")
-        self.requires("boost/1.75.0")
+        if self.options.enable_cpp_wrapper:
+            self.requires("boost/1.75.0")
 
     def validate(self):
         if self.settings.compiler == "Visual Studio" and self.options.use_streaming:
             raise ConanInvalidConfiguration("VS doesn't not compile with enabled option use_streaming")
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
```

Comment:
```suggestion
        copy(self, "COPYING.LIB", self.source_folder, os.path.join(self.package_folder, "licenses"))
        if self.options.build_eeprom_tool:
            copy(self, "COPYING.GPL", self.source_folder, os.path.join(self.package_folder, "licenses"))
```

Looks like the COPYING.* are the full text licenses.